canonical-web-and-design/maas-ui
# feat(ui): set up known_boot_architectures in general state

## Done

- Added `knownBootArchitectures` to general state, and fetched in Edit Boot Architectures form

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to a subnet's details page
- Click "Take action" then click "Edit boot architectures"
- Open Redux DevTools and check that `general.knownBootArchitectures.data` has been populated

## Fixes

Fixes canonical-web-and-design/app-tribe#725
